mw-webapp/src/component/table/Table.tsx

```diff
@@ -1,40 +1,34 @@
-import {useEffect, useState} from "react";
-import {
-  flexRender,
-  getCoreRowModel,
-  useReactTable,
-} from "@tanstack/react-table";
-import {columns} from "src/component/table/columns";
-import {getDayReports} from "src/dataAccessLogic/getDayReports";
+import {memo} from "react";
+import {flexRender} from "@tanstack/react-table";
 import {DayReport} from "src/model/businessModel/DayReport";
+import {ReportsTableProps} from "src/pages/reportsTable/ReportsTable";
 import styles from "src/component/table/Table.module.scss";
 
 /**
- * Table (need update for split component and logic code)
+ * Table's props
  */
-export const Table = () => {
-  const [data, setData] = useState<DayReport[]>([]);
+interface TableProps<T> {
+  /**
+  * Table's data
+  */
+  data: ReportsTableProps<T>;
+}
 
-  const loadDayReports = async () => {
-    const dayReports = await getDayReports();
-    setData(dayReports);
-  };
-
-  useEffect(() => {
-    loadDayReports();
-  }, []);
-
-  const table = useReactTable({
-    data,
-    columns,
-    getCoreRowModel: getCoreRowModel(),
-  });
+/**
+ * Table's model
+ */
+type TablesModel = DayReport
 
+/**
+ * Table
+ */
+export const Table = memo(<T extends TablesModel >(props: TableProps<T>) => {
```

Comment:
Can you please explain why we need memo in this compnent? It's just for me, thanks)

Reply:
Because call methods getRowModel execute extra rerender table.

Reply:
Because call methods rowModel execute extra rerender Table component

Reply:
In the future maybe